cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -1477,18 +1477,25 @@ def _validate_spatial_cell_type_ontology_term_id(self):
 
         :rtype none
         """
-        # Exit if:
-        # - not Visium and is_single is True as no further checks are necessary
-        # - in_tissue is not specified as checks are dependent on this value
-        if not self._is_visium_including_descendants() and self._is_single() or "in_tissue" not in self.adata.obs:
+        self._is_visium_including_descendants()
```

Comment:
it might make sense to just put all of these calls further up the validator, and then rely on `self.is_visium` or `self.is_spatial` states directly instead of needing to call these throughout. feel free to refactor this separately though

Reply:
I agree. However, I don't see a performance issue given that it is only computed if the state is not already set.

Reply:
yeah i think it's fine, it's just slightly confusing to follow. but honestly this entire codebase is so 🤷‍♀️

Reply:
I've created a "#tech" issue for this here: #1125